pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -681,7 +687,7 @@ func (cloud *CloudProvider) waitForZonalOp(ctx context.Context, opName string, z
 	// The v1 API can query for v1, alpha, or beta operations.
 	svc := cloud.service
 	project := cloud.project
-	return wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
+	return wait.Poll(OpInterval, OpTimeout, func() (bool, error) {
```

Comment:
Should we honor the client context.Timeout value instead?

Reply:
Attacher side set context timeout to 15 second. https://github.com/kubernetes-csi/external-attacher/blob/298cb965f909f76f2d2dfa2062cb5e142f14b9d5/cmd/csi-attacher/main.go#L58

my understanding is this context timeout will be used for each individual operation for calling gce api. So we need use another set up timeout for this polling. 
```
op, err := cloud.service.Instances.AttachDisk(cloud.project, instanceZone, instanceName, attachedDiskV1).Context(ctx).Do()
pollOp, err := cloud.service.RegionOperations.Get(cloud.project, region, opName).Context(ctx).Do()
```

Reply:
We override the attacher timeout: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/deploy/kubernetes/base/controller/controller.yaml#L55

Reply:
ah, that makes sense now. The test failure I am checking, the first operation on AttachDisk call timed out after 250 seconds. It is not created successfully at all (not sure about the reason. If it is caused by rate limit, will it error back immediately?). 

So just change here is not enough, we need to reduce the timeout defined in controller.yaml. I think 250s is too long based on normal volume operation. I propose to use 120 seconds or even less.

Reply:
I think adjusting the sidecar timeout based off of the metrics we have is fine. 

But I think the waitForOp method here still needs to use the csi context's timeout instead of a hardcoded one.

Reply:
I can change to use csi context timeout, but the poll interval will still be hardcoded for now.